apps/files/src/plugins/search/folderSearch.ts

```diff
@@ -0,0 +1,70 @@
+import Vue from 'vue'
+
+import { getRequestToken } from '@nextcloud/auth'
+import { emit } from '@nextcloud/event-bus'
+import { translate, translatePlural } from '@nextcloud/l10n'
+import { imagePath } from '@nextcloud/router'
+import type { Node } from '@nextcloud/files'
+
+import '@nextcloud/dialogs/style.css'
+
+import { getFilePickerBuilder } from '@nextcloud/dialogs';
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+(function(OC: any, OCA: any, t: any) {
+
+	/**
+	 *
+	 */
+	function init() {
+		if (!OCA.UnifiedSearch) {
+			return
+		}
+		console.debug('Initializing unified search plugin: folder search from files app')
+		OCA.UnifiedSearch.registerFilterAction({
+			id: 'files',
+			appId: 'files',
+			label: t('files', 'In Folder'),
+			icon: imagePath('files', 'app.svg'),
+			callback: () => {
+				const filepicker = getFilePickerBuilder('Pick plain text files')
+					.addMimeTypeFilter('httpd/unix-directory')
+					.allowDirectories(true)
+					.addButton({
+						label: 'Pick',
+						callback: (nodes: Node[]) => {
+							console.debug('Folder picked', nodes[0])
+							const folder = nodes[0]
+							emit('nextcloud:unified-search:add-filter', {
+								id: 'files',
+								payload: folder,
+								filterUpdateText: t('files', 'Search in folder: {folder}', { folder: folder.attributes.filename }),
+								filterParams: { path: folder.attributes.filename },
```

Comment:
```suggestion
								filterParams: { path: folder.dirname },
```

https://nextcloud-libraries.github.io/nextcloud-files/classes/File.html#dirname